### DISCUSSION TITLE
Similar functionality's to Ocelot's "DownstreamHeaderTransform"

### DISCUSSION THREAD

**Discussion by neptun09** [dotnet/yarp#1410] [Q&A] [answered]
Hi,

is it possible to have something similar to functionality called "DownstreamHeaderTransform" in Ocelot. In Ocelot you can define below sample rule:

`"DownstreamHeaderTransform": {
    "Test": 
"http://www.bbc.co.uk/, http://ocelot.com/"
}`

so all incoming request using url: "http://www.bbc.co.uk" will be replaced with "http://ocelot.com/".

What is the best way to achieve similar functionality in YARP? 

Best regards

**Comment by Kahbazi** [comment, marked as answer]
There's no built-in configuration to replace request headers value but you can add your own transform. Something like this :

```csharp
builder.Services
    .AddReverseProxy()
    .AddTransforms(transforms =>
    {
        // transforms.Cluster.Metadata 
        // get values from cluster metadata

        transforms.AddRequestTransform(context =>
        {
            if (context.ProxyRequest.Headers.TryGetValues("Test", out var value) && value.FirstOrDefault() == "http://www.bbc.co.uk/")
            {
                context.ProxyRequest.Headers.Remove("Test");
                context.ProxyRequest.Headers.Add("Test", "http://ocelot.com/");
            }
            return default;
        });
    });
```

**Comment by neptun09** [comment]
Thanks a lot for quick answear. Indeed, the provided sample give me a hint how to implement it :)

**Reply by Tratcher** [reply]
Can you give an example where a request header would contain a url like that? Those values usually go into other fields.
